src/track/tracktiminginfo.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <QObject>
+
+#include "track/track.h"
+#include "track/trackplaytimers.h"
+
+/// Manages different timing infos required for the scrobbling
+/// services
+class TrackTimingInfo : public QObject {
+    Q_OBJECT
+  public:
+    TrackTimingInfo(TrackPointer pTrack);
+    void pausePlayedTime();
+    void resumePlayedTime();
+    void resetPlayedTime();
+    void setElapsedTimer(TrackTimers::ElapsedTimer* elapsedTimer);
+    void setTimer(TrackTimers::RegularTimer* timer);
+    void setMsPlayed(qint64 ms);
+    bool isScrobbable() const;
+    void setTrackPointer(TrackPointer pTrack);
+    bool isTimerPaused() const;
+  public slots:
+    void slotCheckIfScrobbable();
+    void slotGuiTick(double timeSinceLastTick);
+  signals:
+    void readyToBeScrobbled(TrackPointer pTrack);
+
+  private:
+    std::unique_ptr<TrackTimers::ElapsedTimer> m_pElapsedTimer;
+    std::unique_ptr<TrackTimers::RegularTimer> m_pTimer;
```

Comment:
RegularTimer is a QObject and should better be managed by parented_ptr. Otherwise it might get deleted while it is still referenced by the event loop, see QObject::deleteLater().